lib/chef/win32/file.rb

```diff
@@ -155,6 +155,27 @@ def self.verify_links_supported!
         end
       end
 
+      def self.file_access_check(path, desired_access)
+        security_descriptor = Chef::ReservedNames::Win32::Security.get_file_security(path)
+        token_rights = Chef::ReservedNames::Win32::Security::TOKEN_IMPERSONATE | 
```

Comment:
I think i need TOKEN_IMPERSONATE and TOKEN_DUPLICATE so that i can duplicate the token. It does not let me use the current processes token as is